### PR TITLE
test-configs.yaml: add kselftest-lib test plan

### PR DESCRIPTION
**Pull request by alexandrasp**
add a test plan to run lib kselftest collection on hip07-d05 device.
Depends on #583 

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

### CODE REVIEW

File: test-configs.yaml [hunk 1]
```diff
@@ -176,12 +176,21 @@ test_plans:
...
     params:
       job_timeout: '120'
+      tst_cmdfiles: ""
+
+  kselftest-lib:
+    <<: *kselftest
+    params:
+      job_timeout: '360'
+      kselftest_collections: lib
+      tst_cmdfiles: "lib"
```
Comment (gctucker) [inline test-configs.yaml:193]: Where is this used?

Comment (alexandrasp) [reply test-configs.yaml:193]: this is not used in kselftest context, thanks :)

**Comment by alexandrasp** [comment]
@gctucker now that the build ran with success applying #583 , could you remove skip tag from this ?

**Comment by gctucker** [comment]
Sure, sorry I missed that part.  However, it looks like this PR is conflicting so it's not getting merged on the staging branch.

**Comment by gctucker** [comment]
Ah it's conflicting with the LTP changes, so I'll put the skip label there.

**Review by gctucker — COMMENTED** [review]
OK, I think we can drop the `kselftest` in test configs as we should be only running the subsets.

Also I don't see any results for kselftest-lib on staging yet, it needs investigating what's going on.

**Comment by gctucker** [comment]
I think you're just missing this bit in the `lab-collabora` section:
```diff
diff --git a/lab-configs.yaml b/lab-configs.yaml
index 9a60ef9006d2..8e0b5b3a9404 100644
--- a/lab-configs.yaml
+++ b/lab-configs.yaml
@@ -58,6 +58,7 @@ labs:
             - igt-kms-tegra
             - igt-gpu-i915
             - igt-gpu-panfrost
+            - kselftest-lib
             - ltp-crypto
             - ltp-ipc
             - sleep
```

**Comment by alexandrasp** [comment]
Tested OK on staging with some devices in collabora and baylibre labs:

https://staging.kernelci.org/test/job/kernelci/branch/staging-next/kernel/staging-next-20210123.0/plan/kselftest-lib/

Tests ran successfully on [rk3288-veyron-jaq](https://staging.kernelci.org/test/plan/id/600be763e4f4a02a1674f907/) and [hip07-d05](https://staging.kernelci.org/test/plan/id/600be78e37db24a5c974f8fe/)

**Comment by gctucker** [comment]
Great, please check if `hana` and `kevin` are booting on `mainline` or if the issue is only on `linux-next`.  If they're also failing on mainline then I think we should drop them from this PR for now until the root case of the boot failures are fixed.

**Comment by alexandrasp** [comment]
> Great, please check if `hana` and `kevin` are booting on `mainline` or if the issue is only on `linux-next`. If they're also failing on mainline then I think we should drop them from this PR for now until the root cause of the boot failures are fixed.

Removed from this PR because they are failing on mainline.

**Comment by gctucker** [comment]
Tested OK on staging:
Tested OK on staging: https://staging.kernelci.org/test/job/kernelci/branch/staging-next/kernel/staging-next-20210125.0/plan/kselftest-lib/

**Review by gctucker — COMMENTED** [review]
I think the commit message should be tweaked as it's not enabling kselftest-lib on all the devices that can run kselftest but only on rk3288-veyron-jaq and hip07-d05.

Other than that, the changes seem fine and it's working as expected on staging.

**Comment by alexandrasp** [comment]
> I think the commit message should be tweaked as it's not enabling kselftest-lib on all the devices that can run kselftest but only on rk3288-veyron-jaq and hip07-d05.
> 
> Other than that, the changes seem fine and it's working as expected on staging.

it makes sense! Commit message updated, thanks @gctucker

**Comment by gctucker** [comment]
There was a problem on staging with no results since last night, I think it's fixed now so we should be able to merge this in a couple of hours when we get a new set of results.

**Comment by gctucker** [comment]
OK it's all back to normal, with at least one platform enabled to run these tests in lab-collabora:
https://staging.kernelci.org/test/job/kernelci/branch/staging-next/kernel/staging-next-20210126.3/plan/kselftest-lib/

**Pull request merged**